### ISSUE TITLE
no-dupe-class-members not warning about static non-method members

### ISSUE THREAD

**Issue by ajmath** [eslint/eslint#7534]
* **ESLint Version:** 3.2.2
* **Node Version:** 6.0.0
* **npm Version:** 3.8.6

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```
rules:
  no-dupe-class-members: 'error'

parser: "babel-eslint"
```

**What did you do? Please include the actual source code causing the issue.**

```js
export default class Foo {
  static bar = 'test';
  static bar = 'test';
}
```

**What did you expect to happen?**
Should have caused
```
/Foo.js
  5:3  error  Duplicate name 'bar'  no-dupe-class-members
```

**What actually happened? Please include the actual, raw output from ESLint.**
`⇒ ./node_modules/eslint/bin/eslint.js --ext .js Foo.js` returns no output

**Comment by vitorbal** [comment]
Thank you for your issue.
Static instance properties are not supported in ESLint yet, so my guess is we are not traversing those nodes for the `no-dupe-class-members` rule.
Since static instance properties are still a stage 2 (i think) proposal, your best bet would be a babel-eslint rule that superseeds the eslint one.

**Comment by mysticatea** [comment]
Yeah, class fields are not part of JavaScript yet.
Also, ESLint handles the AST which according to [ESTree specification](https://github.com/estree/estree), but ESTree does not include class fields.

Maybe You can propose a rule for this to [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel#readme).

**Comment by pastelsky** [comment]
Now that class properties have moved to stage-3, is eslint considering supporting this?

**Comment by not-an-aardvark** [comment]
As described in our [readme](https://github.com/eslint/eslint/blob/61f109343503c13de7b742669e5df44964c9e4e7/README.md#what-about-experimental-features), we only support stage 4 features.
